### PR TITLE
Version5 support

### PR DESCRIPTION
**Pull request by brbog**
Mostly documentation effort.

Authentication objects shouldn't throw checked exceptions on creation. Will fail as early as possible, and must probably be fixed manually anyway.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,13 +9,34 @@ crawling the Web. Using it, you can setup a multi-threaded web crawler in few mi
...
+Some highlights:
+
+- choice of frontiers => avoid using a database with a license that doesn't comply with your use-case
+- easily substituting various parser implementations (not only for html, but also css, binary, and plain text)
+- dynamic authentication
+- improved exception handling, more versatile to customize
+- fixes various parsing issues
+- more documentation
+- more tests (JUnit)
+- uses Maven
```
Comment (rzo1) [inline README.md:35]:
```suggestion
- uses Apache Maven as build tool
```

File: README.md [hunk 2]
```diff
@@ -9,13 +9,34 @@ crawling the Web. Using it, you can setup a multi-threaded web crawler in few mi
...
 - [More Examples](#more-examples)
 - [Configuration Details](#configuration-details)
 - [Reconstructing extra urls to crawl](#reconstructing-extra-urls-to-crawl)
+- [Authentication](#authentication)
+- [High-level design diagrams](#high-level-design-diagrams)
 - [License](#license)
 
+## Why use this fork?
+
+This fork starts where development on the previous main repository stalled.
```
Comment (rzo1) [inline README.md:24]:
```suggestion
This fork starts where the development of the previous main repository stalled.
```

File: pom.xml [hunk 3]
```diff
@@ -65,6 +65,15 @@
             <url>https://github.com/rzo1/crawler4j</url>
             <timezone>Europe/Berlin</timezone>
         </developer>
+        <developer>
```
Comment (rzo1) [inline pom.xml:68]: :+1:

File: README.md [hunk 4]
```diff
@@ -9,13 +9,34 @@ crawling the Web. Using it, you can setup a multi-threaded web crawler in few mi
 
 ## Table of content
 
+- [Why use this fork?](#why-use-this-fork)
```
Comment (rzo1) [inline README.md:12]:
```suggestion
- [Why you should use this fork?](#why-you-should-use-this-fork)
```

File: README.md [hunk 5]
```diff
@@ -9,13 +9,34 @@ crawling the Web. Using it, you can setup a multi-threaded web crawler in few mi
...
 - [Installation](#installation)
 - [Quickstart](#quickstart)   
 - [More Examples](#more-examples)
 - [Configuration Details](#configuration-details)
 - [Reconstructing extra urls to crawl](#reconstructing-extra-urls-to-crawl)
+- [Authentication](#authentication)
+- [High-level design diagrams](#high-level-design-diagrams)
 - [License](#license)
 
+## Why use this fork?
```
Comment (rzo1) [inline README.md:22]:
```suggestion
## Why you should use this fork?
```

File: README.md [hunk 6]
```diff
@@ -9,13 +9,34 @@ crawling the Web. Using it, you can setup a multi-threaded web crawler in few mi
...
+
+This fork starts where development on the previous main repository stalled.
+
+Some highlights:
```
Comment (rzo1) [inline README.md:26]:
```suggestion
Some highlights include:
```

File: README.md [hunk 7]
```diff
@@ -9,13 +9,34 @@ crawling the Web. Using it, you can setup a multi-threaded web crawler in few mi
...
+
+- choice of frontiers => avoid using a database with a license that doesn't comply with your use-case
+- easily substituting various parser implementations (not only for html, but also css, binary, and plain text)
```
Comment (rzo1) [inline README.md:29]:
```suggestion
- easy substitution of various parser implementations (not only for html, but also css, binary, and plain text)
```

File: README.md [hunk 8]
```diff
@@ -9,13 +9,34 @@ crawling the Web. Using it, you can setup a multi-threaded web crawler in few mi
...
+- [High-level design diagrams](#high-level-design-diagrams)
 - [License](#license)
 
+## Why use this fork?
+
+This fork starts where development on the previous main repository stalled.
+
+Some highlights:
+
+- choice of frontiers => avoid using a database with a license that doesn't comply with your use-case
```
Comment (rzo1) [inline README.md:28]:
```suggestion
- choice between multiple frontier implementation => avoid using a database with a license that doesn't comply with your use-case
```

File: README.md [hunk 9]
```diff
@@ -9,13 +9,34 @@ crawling the Web. Using it, you can setup a multi-threaded web crawler in few mi
...
+- easily substituting various parser implementations (not only for html, but also css, binary, and plain text)
+- dynamic authentication
+- improved exception handling, more versatile to customize
+- fixes various parsing issues
+- more documentation
+- more tests (JUnit)
+- uses Maven
+- provides clean upgrade paths by keeping backward compatibility in mind and deprecating methods before removing them
```
Comment (rzo1) [inline README.md:36]:
```suggestion
- provides a clean upgrade paths by keeping backward compatibility in mind and deprecating methods before removing them
```

File: README.md [hunk 10]
```diff
@@ -9,13 +9,34 @@ crawling the Web. Using it, you can setup a multi-threaded web crawler in few mi
...
+- more eyes have gone through the code, so readability and correctness have improved
+- maintained, so dependencies are often updated to their latest versions
```
Comment (rzo1) [inline README.md:38]:
```suggestion
- maintained, i.e. dependencies are often updated to their latest versions
```

**Review by rzo1 — APPROVED** [review]

**Comment by rzo1** [comment]
I have only some suggestions, which are no **must** have. If you like, you can go through it and notify me, if you think, that we are good to go here.

**Comment by brbog** [comment]
@rzo1 : good to go!

**Pull request merged**